src/__internal__/popover/popover.test.tsx

```diff
@@ -0,0 +1,159 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import Popover, { PopoverProps } from "./popover.component";
+import { baseTheme } from "../../style/themes";
+
+const PopoverWithButton = (
+  props: Omit<PopoverProps, "children" | "reference">
+) => {
+  const buttonRef = React.useRef<HTMLButtonElement>(null);
+
+  return (
+    <>
+      <button ref={buttonRef} type="button">
+        Reference
+      </button>
+      <Popover reference={buttonRef} {...props}>
+        <div>I float!</div>
+      </Popover>
+    </>
+  );
+};
+
+test("popup content is visible by default", () => {
+  render(
+    <Popover reference={{ current: null }}>
+      <div>I float!</div>
+    </Popover>
+  );
+
+  expect(screen.getByText("I float!")).toBeVisible();
+});
+
+test("popup content is visible when isOpen prop is true", () => {
+  render(
+    <Popover isOpen reference={{ current: null }}>
+      <div>I float!</div>
+    </Popover>
+  );
+
+  expect(screen.getByText("I float!")).toBeVisible();
+});
+
+test("popup content is hidden when isOpen prop is false", () => {
+  render(
+    <Popover isOpen={false} reference={{ current: null }}>
+      <div>I float!</div>
+    </Popover>
+  );
+
+  expect(screen.getByText("I float!")).not.toBeVisible();
+});
+
+test("renders popup in the document body to avoid any ancestors affecting its layout and styling", () => {
+  render(
+    <main>
+      <PopoverWithButton />
+    </main>
+  );
+
+  expect(document.body).toHaveTextContent(/I float!/);
+  expect(screen.getByRole("main")).not.toHaveTextContent(/I float!/);
+});
+
+test("renders popup within the component's ancestor as normal when disablePortal prop is true", () => {
+  render(
+    <main>
+      <PopoverWithButton disablePortal />
+    </main>
+  );
+
+  expect(screen.getByRole("main")).toHaveTextContent(/I float!/);
+});
+
+test("when mounted, renders popup in the reference's nearest dialog ancestor if it has one", async () => {
+  const DialogWithPopover = () => {
+    const [isOpen, setIsOpen] = React.useState(false);
+    const buttonRef = React.useRef<HTMLButtonElement>(null);
+
+    return (
+      <dialog open>
+        <button
+          ref={buttonRef}
+          type="button"
+          onClick={() => setIsOpen((prev) => !prev)}
+        >
+          Toggle popover
+        </button>
+        {isOpen && (
+          <Popover reference={buttonRef}>
+            <div>I float!</div>
+          </Popover>
+        )}
+      </dialog>
+    );
+  };
+  const user = userEvent.setup();
+  render(<DialogWithPopover />);
+
+  // mount the popover
```

Comment:
**question** (non-blocking): not a big deal, but I'm curious why in this test we're using a user action (button click) to trigger the popover opening, when in the previous tests we are simply making assertions about an always-open popover. This test is only about where in the DOM it is rendered, which shouldn't be affected by precisely when the component was mounted.

Reply:
Conditional rendering of the `Popover` is necessary here because without it, the button ref passed to reference remains uninitialised, causing the test to fail. While I agree that refactoring how the component handles React refs would be a cleaner long-term solution, I opted for this approach to maintain consistency with the previous implementation and ensure the test passes for now.

Reply:
Yeah I too feel this is a bug, I'll raise a ticket for this 👍🏼

Reply:
@robinzigmond Raised as **FE-6819**